src/bb_excel/core.clj

```diff
@@ -132,68 +133,74 @@
 
 (defn process-cell
   "Process Excel cell"
-  [dict styles coll]
-  (let [[[_ row col]] (re-seq #"([A-Z]*)([0-9]+)" (:r coll))
-        u (-> coll
-              (assoc :x row)
-              (assoc :y col))]
+  [dict styles cell]
+  (let [[_ row-index col-index] (re-matches #"([A-Z]+)([0-9]+)" (:r cell))
+        cell* (merge cell
+                     {:x row-index
+                      :y col-index})
+        cell-type (:t cell*)
+        cell-value (:d cell*)]
     (cond
-      ;; Possible data types well explained here https://stackoverflow.com/a/18346273
-      (= (:t u) "s")                (dissoc (assoc-in u [:d] (dict (read-string (:d u)))) :t)
-      (= (:t u) "str")              (dissoc u :t)
-      (= (:t u) "inlineStr")        (dissoc u :t)
-      (= (:t u) "b")                (dissoc (assoc-in u [:d] (if (= "1" (:d u)) true false)) :t)
-      (= (:t u) "e")                (assoc-in u [:d] (error-codes (:d u)))
-      (= (:t u) "n")                (assoc u :d (parse-long (:d u)))
-      (style-check u styles pcts)   (assoc-in u [:d] (num2pct (:d u)))
-      (style-check u styles dates)  (assoc-in u [:d] (num2date (:d u)))
-      (style-check u styles times)  (assoc-in u [:d] (num2time (:d u)))
-      :else u)))
+      ;; Possible cell-value types well explained here https://stackoverflow.com/a/18346273
+      (= cell-type "s")                (assoc cell* :d (get dict (parse-long cell-value)))
+      (= cell-type "str")              cell*
+      (= cell-type "inlineStr")        cell*
+      (= cell-type "b")                (assoc cell* :d (if (= "1" cell-value) true false))
+      (= cell-type "e")                (assoc cell* :d (get error-codes cell-value))
+      (= cell-type "n")                (assoc cell* :d (parse-long cell-value))
+      (style-check cell* styles pcts)   (assoc-in cell* [:d] (num2pct cell-value))
+      (style-check cell* styles dates)  (assoc-in cell* [:d] (num2date cell-value))
+      (style-check cell* styles times)  (assoc-in cell* [:d] (num2time cell-value))
+      :else cell*)))
+
+(defn- get-row-index
+  [row]
+  (parse-long (:y row)))
 
 (defn process-row
   "Process Excel row of data"
-  [dict styles coll]
-  (reduce #(merge % {:_r (read-string (:y %2)) (keyword (:x %2)) (:d %2)}) {}
-          (map (partial process-cell dict styles)
-               (map #(merge (first %) {:d (second %)}
-                            {:f (nth % 2)})
-                    (map (juxt :attrs
-                               (comp last :content last :content)
-                               (comp first :content first :content)) coll)))))
+  [dict styles row]
+  (let [row* (->> row
+                  (map (fn [cell] (merge (:attrs cell)
+                                         {:d (-> cell :content last :content last)})))
+                  (map (partial process-cell dict styles)))
+        row-index (get-row-index (first row*))]
+    (into {:_r row-index} (map #(-> [(keyword (:x %)) (:d %)]))
```

Comment:
#(-> <some new data form>) maybe not clear syntax, but it's just response with new data form